docs/tutorials/control/flutter-app.md

```diff
@@ -0,0 +1,736 @@
+---
+title: "Build a Flutter App that Integrates with Viam"
+linkTitle: "Build a Flutter App"
+type: "docs"
+description: "Use Viam's Flutter SDK to build a custom mobile app to show your machines and their components."
+webmSrc: "/tutorials/flutter-app/demo.webm"
+mp4Src: "/tutorials/flutter-app/demo.mp4"
+videoAlt: "An example Viam-integrated Flutter app."
+tags: ["sdk", "flutter"]
+authors: ["Clint Purser"]
+languages: ["flutter"]
+viamresources: ["base"]
+level: "Intermediate"
+date: "2024-01-17"
+cost: "0"
+---
+
+<div class="td-max-width-on-larger-screens">
+ <div class="alignleft" style="max-width:250px; margin-right:40px">
+  {{<gif webm_src="/tutorials/flutter-app/demo.webm" mp4_src="/tutorials/flutter-app/demo.mp4" alt="Rendering of the mobile app. Log in is clicked, then all Clint's locations are shown, Clint's Desk it clicked, all its smart machines are listed, and then desk-robot is clicked, showing a list of components and services belonging to that smart machine.">}}
+  </div>
+</div>
+
+Flutter is Google's user interface toolkit for building applications for mobile, web, and desktop from a single codebase.
+If the existing [Viam mobile app](/fleet/#the-viam-mobile-app) doesn't fit your needs, you can use Viam's [Flutter SDK](https://flutter.viam.dev/) to build a custom app to interact with your smart machines that run on Viam.
+
+This tutorial guides you through creating a mobile app that shows your machines and their components.
+As you work through this project you will learn the following:
+
+- The basics of how Flutter works
+- How to create _layouts_ in Flutter
+- How to connect user interactions (such as button presses) to app behavior
+- The basics of using Viam's Flutter SDK
+
+## Requirements
+
+You do not need any hardware for this tutorial other than a computer running macOS or a 64-bit Linux operating system.
+
+This tutorial assumes you already have a smart machine [configured](/build/configure/) on the [Viam app](https://app.viam.com).
+
+## Set up your Flutter environment
+
+To make this documentation as straightforward as possible, this tutorial assumes you are using [Visual Studio Code](https://code.visualstudio.com/download) (VS Code) as your development environment (IDE).
```

Comment:
Ooooh. I'll apply all your feedback and then look at rewriting.

Copying aside, to the original comment here, I would say your suggestion changes the meaning; part of why we're saying that they should use VSC is so that we can tell them exactly which buttons to push etc., not just for the boilerplate.
```suggestion
This tutorial uses [Visual Studio Code](https://code.visualstudio.com/download) (VS Code) as the development environment (IDE), and uses the VS Code [Flutter extension](https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter) to generate necessary boilerplate code.
```